funnel/views/label.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+
+from flask import flash, redirect, g, request
+from werkzeug.datastructures import MultiDict
+from coaster.views import render_with, requires_permission, route, UrlForView, ModelView
+from baseframe import _, forms
+from baseframe.forms import render_delete_sqla
+
+from .. import app, funnelapp, lastuser
+from ..models import Label, db, Project, Profile
+from ..forms import LabelForm, LabelOptionForm
+from .mixins import ProjectViewMixin
+from .decorators import legacy_redirect
+
+
+@route('/<profile>/<project>/labels')
+class ProjectLabelView(ProjectViewMixin, UrlForView, ModelView):
+    __decorators__ = [legacy_redirect]
+
+    @route('', methods=['GET', 'POST'])
+    @render_with('labels.html.jinja2')
+    @lastuser.requires_login
+    @requires_permission('edit_project')
+    def labels(self):
+        form = forms.Form()
+        if form.validate_on_submit():
+            idlist = request.values.getlist('id')
+            for idx, lid in enumerate(idlist, start=1):
+                lbl = Label.query.get(lid)
+                if lbl is not None:
+                    lbl.seq = idx
+                    db.session.commit()
+            flash(_("Your changes have been saved"), category='success')
+        return dict(project=self.obj, labels=self.obj.labels, form=form)
+
+    @route('new', methods=['GET', 'POST'])
+    @lastuser.requires_login
+    @render_with('labels_form.html.jinja2')
+    @requires_permission('admin')
+    def new_label(self):
+        form = LabelForm(model=Label, parent=self.obj.parent)
+        emptysubform = LabelOptionForm(MultiDict({}))
+        if form.validate_on_submit():
+            # This form can send one or multiple values for title and icon_emoji.
+            # If the label doesn't have any sublabel, one value is sent for each list,
+            # and those values are also available at `form.data`.
+            # But in case there are sublabels, the sublabel values are in the list
+            # in the order they appeared on the create form.
+            titlelist = request.values.getlist('title')
+            emojilist = request.values.getlist('icon_emoji')
```

Comment:
Use `zip` for this. It will make the code clearer. https://www.geeksforgeeks.org/zip-in-python/

Reply:
zip doesn't make much clean as I need to use `enumerate` for sequence. even if I zip, i have to unzip after enumerate.